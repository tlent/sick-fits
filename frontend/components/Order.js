import React, { Component } from "react";
import PropTypes from "prop-types";
import { Query } from "react-apollo";
import { format } from "date-fns";
import Head from "next/head";
import gql from "graphql-tag";
import formatMoney from "../lib/formatMoney";
import ErrorMessage from "./ErrorMessage";
import OrderStyles from "./styles/OrderStyles";

const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!) {
    order(orderID: $id) {
      id
      charge
      total
      createdAt
      user {
        id
      }
      items {
        id
        title
        description
        price
        image
        quantity
      }
    }
  }
`;

class Order extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired
  };
  render() {
    return (
      <Query query={SINGLE_ORDER_QUERY} variables={{ id: this.props.id }}>
        {({ data, loading, error }) => {
          if (error) return <ErrorMessage error={error} />;
          if (loading) return <p>Loading...</p>;
          const { order } = data;
          return (
            <OrderStyles>
              <Head>
                <title>Sick Fits | Order {order.id}</title>
              </Head>
              <p>
                <span>Order ID</span>
                <span>{order.id}</span>
              </p>
              <p>
                <span>Charge</span>
                <span>{order.charge}</span>
              </p>
              <p>
                <span>Date</span>
                <span>{format(order.createdAt, "MMMM d, yyyy h:mm a")}</span>
              </p>
              <p>
                <span>Order Total</span>
                <span>{formatMoney(order.total)}</span>
              </p>
              <p>
                <span>Unique Item Count</span>
                <span>{order.items.length}</span>
              </p>
              <div className="items">
                {order.items.map(item => (
                  <div className="order-item" key={item.id}>
                    <img src={item.image} alt={item.title} />
                    <div className="item-details">
                      <h2>{item.title}</h2>
                      <p>Count: {item.quantity}</p>
                      <p>Price: {formatMoney(item.price)}</p>
                      <p>Subtotal: {formatMoney(item.price * item.quantity)}</p>
                      <p>{item.description}</p>
                    </div>
                  </div>
                ))}
              </div>
            </OrderStyles>
          );
        }}
      </Query>
    );
  }
}

export default Order;
