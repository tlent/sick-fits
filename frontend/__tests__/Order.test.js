import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import Order, { SINGLE_ORDER_QUERY } from "../components/Order";
import { MockedProvider } from "react-apollo/test-utils";
import { fakeOrder } from "../lib/testUtils";

const mocks = [
  {
    request: { query: SINGLE_ORDER_QUERY, variables: { id: "abc123" } },
    result: { data: { order: { ...fakeOrder(), id: "abc123" } } }
  }
];

describe("Order component", () => {
  it("renders and matches snapshot", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Order id={"abc123"} />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find("div[data-test='Order']"))).toMatchSnapshot();
  });
});
