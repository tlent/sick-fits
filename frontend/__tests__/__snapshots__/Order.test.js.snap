// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component renders and matches snapshot 1`] = `
<div
  className="OrderStyles-sc-4oqalm-0 camMij"
  data-test="Order"
>
  <SideEffect(Head)>
    <Head />
  </SideEffect(Head)>
  <p>
    <span>
      Order ID
    </span>
    <span>
      abc123
    </span>
  </p>
  <p>
    <span>
      Charge
    </span>
    <span>
      ch_123
    </span>
  </p>
  <p>
    <span>
      Date
    </span>
    <span>
      March 31, 2018 8:00 PM
    </span>
  </p>
  <p>
    <span>
      Order Total
    </span>
    <span>
      $400
    </span>
  </p>
  <p>
    <span>
      Unique Item Count
    </span>
    <span>
      2
    </span>
  </p>
  <div
    className="items"
  >
    <div
      className="order-item"
      key="3a430a73-c9e9-4bcc-b46a-41618965ffea"
    >
      <img
        alt="soluta non omnis consequatur enim quia autem"
        src="reprehenderit.jpg"
      />
      <div
        className="item-details"
      >
        <h2>
          soluta non omnis consequatur enim quia autem
        </h2>
        <p>
          Count: 
          1
        </p>
        <p>
          Price: 
          $42.34
        </p>
        <p>
          Subtotal: 
          $42.34
        </p>
        <p>
          et modi tenetur amet modi reprehenderit omnis
        </p>
      </div>
    </div>
    <div
      className="order-item"
      key="2393e090-592f-4d03-b808-c9d81098deec"
    >
      <img
        alt="quia sed exercitationem omnis laborum exercitationem est"
        src="sint.jpg"
      />
      <div
        className="item-details"
      >
        <h2>
          quia sed exercitationem omnis laborum exercitationem est
        </h2>
        <p>
          Count: 
          1
        </p>
        <p>
          Price: 
          $42.34
        </p>
        <p>
          Subtotal: 
          $42.34
        </p>
        <p>
          sapiente laudantium molestias assumenda quasi adipisci mollitia
        </p>
      </div>
    </div>
  </div>
</div>
`;
